### PR TITLE
Implement Elytra (old)

### PR DESCRIPTION
**Pull request by pandaaaBE**
## Introduction
Implement Elytra.

_There is a plugin, why should we accept your pull request?
Well, since IvanCrafter is currently working on a Fireworks PR it would be only fitting to finally add the Elytra as well - it's tedious to search out all the vanilla features on poggit._

## Todo
- [x] Add Repairing via Crafting Recipes
- [x] Disable when touching ground

## Tests
[Video](https://streamable.com/iefpxd)

## Facts
This is my first pull request so please don't laugh at me, I will of course add any features I find at minecraft.wiki.

### CODE REVIEW

**Comment by pandaaaBE** [comment]
1. Is there something better then "onTickWorn"?
2. Is the ElytraLoopSound class relevant or is this useless?

**Comment by pandaaaBE** [comment]
Should I add repairing via crafting table for elytras? I can not find a method for this

File: src/item/Elytra.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+			// Check if the player's pitch is within the allowed range between -59 and 38.
+			if ($pitch >= -59 && $pitch <= 38) {
+				$entity->resetFallDistance();
+				return true;
+			}
+
+			// If the player is on the ground, reset the fall distance and disable gliding.
+			if ($entity->isOnGround()) {
+				$entity->resetFallDistance();
```
Comment (JavierLeon9966) [inline src/item/Elytra.php:46]: Why is this here? Don't we already reset fall damage above?

File: src/item/Elytra.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+	public function onTickWorn(Living $entity): bool {
+		if ($entity instanceof Human && $entity->isGliding()) {
+			$this->applyDamage(1);
+
+			$location = $entity->getLocation();
+			$pitch = $location->pitch;
+
+			// Check if the player's pitch is within the allowed range between -59 and 38.
+			if ($pitch >= -59 && $pitch <= 38) {
```
Comment (JavierLeon9966) [inline src/item/Elytra.php:39]: I think this should be constants or use a calculation to check if it'll reset fall damage.

File: src/item/VanillaItems.php [hunk 3]
```diff
@@ -626,6 +627,7 @@ private static function registerArmorItems() : void{
 		self::register("iron_chestplate", new Armor(new IID(Ids::IRON_CHESTPLATE), "Iron Chestplate", new ArmorTypeInfo(6, 241, ArmorInventory::SLOT_CHEST, material: ArmorMaterials::IRON()), [EnchantmentTags::CHESTPLATE]));
 		self::register("leather_tunic", new Armor(new IID(Ids::LEATHER_TUNIC), "Leather Tunic", new ArmorTypeInfo(3, 81, ArmorInventory::SLOT_CHEST, material: ArmorMaterials::LEATHER()), [EnchantmentTags::CHESTPLATE]));
 		self::register("netherite_chestplate", new Armor(new IID(Ids::NETHERITE_CHESTPLATE), "Netherite Chestplate", new ArmorTypeInfo(8, 593, ArmorInventory::SLOT_CHEST, 3, true, material: ArmorMaterials::NETHERITE()), [EnchantmentTags::CHESTPLATE]));
+		self::register("elytra", new Elytra(new IID(Ids::ELYTRA), "Elytra", new ArmorTypeInfo(0, 433, ArmorInventory::SLOT_CHEST, 0, false, null)));
```
Comment (JavierLeon9966) [inline src/item/VanillaItems.php:630]: You can skip the optional parameters and the elytra class must override the `onBroken()` with an empty one to prevent it from breaking.

File: src/world/sound/ElytraLoopSound.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+declare(strict_types=1);
+
+namespace pocketmine\world\sound;
+
+use pocketmine\math\Vector3;
+use pocketmine\network\mcpe\protocol\LevelEventPacket;
+use pocketmine\network\mcpe\protocol\types\LevelEvent;
+
+class ElytraLoopSound implements Sound{
```
Comment (JavierLeon9966) [inline src/world/sound/ElytraLoopSound.php:30]: Isn't this sound client side?

**Review by JavierLeon9966 — CHANGES_REQUESTED** [review]
This needs to be in the minor-next branch

**Comment by JavierLeon9966** [comment]
> Should I add repairing via crafting table for elytras? I can not find a method for this

That can be skipped. Dynamic crafting hasn't been added yet.

File: src/item/Elytra.php [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+namespace pocketmine\item;
+
+use pocketmine\entity\Human;
+use pocketmine\entity\Living;
+
+class Elytra extends Armor{
+
+	public function onTickWorn(Living $entity): bool {
+		if ($entity instanceof Human && $entity->isGliding()) {
+			$this->applyDamage(1);
```
Comment (JavierLeon9966) [inline src/item/Elytra.php:33]: Damage should only be applied every second.

File: src/world/sound/ElytraLoopSound.php (hunk 4 above)
Comment (pandaaaBE) [reply src/world/sound/ElytraLoopSound.php:30]: I think so, this is why I was asking.

Will remove the class

File: src/item/Elytra.php (hunk 2 above)
Comment (pandaaaBE) [reply src/item/Elytra.php:39]: How do you mean?

**Review by JavierLeon9966 — CHANGES_REQUESTED** [review]

File: src/item/Elytra.php (hunk 2 above)
Comment (JavierLeon9966) [reply src/item/Elytra.php:39]: You shouldn't use hardcoded values.
